nemo/core/classes/modelPT.py

```diff
@@ -177,12 +178,39 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None):
                     f"Test config : \n{OmegaConf.to_yaml(self._cfg.test_ds)}"
                 )
 
+        # Create list of lists for val and test outputs to support multiple dataloaders
```

Comment:
Oh now i see - they removed the `outputs` arg for validation_epoch_end and test_epoch_end so we need a cache storage inside the model..

Btw - this if check - it will fail to execute for all Megatron models + any model that adds `defer_setup=True` in their val/test_ds configs. We need to add these checks below as well somewhere.

Reply:
Yes you are right, for cases where setup is not happening in ModelPT and multiple dataloaders were getting used, the tests gave error and I have added this check in those specific model files. Since they were cases where setup was overridden inside the model class. 
Any suggestions on if we want to add this condition check to all models that have setup overridden or add it when the need arises ?
I have added the check again in ModelPT inside def setup() to handle scenario when defere_setup is True.

Reply:
Discussed offline to add on_train_start, on_validation_start hooks to ModelPT and perform the check for multi dataloaders in the hooks. Deferring it to a subsequent PR.